search/signals.py

```diff
@@ -22,37 +28,50 @@
 # otherwise it would run for any `post_save`/`post_delete` coming from any model
 
 
+# pylint: disable=unused-argument
 @receiver(post_save, sender=Profile, dispatch_uid="profile_post_save_index")
-def handle_update_profile(sender, instance, **kwargs):  # pylint: disable=unused-argument
+def handle_update_profile(sender, instance, **kwargs):
     """Update index when Profile model is updated."""
     index_users.delay([instance.user])
 
 
 @receiver(post_save, sender=Education, dispatch_uid="education_post_save_index")
-def handle_update_education(sender, instance, **kwargs):  # pylint: disable=unused-argument
+def handle_update_education(sender, instance, **kwargs):
     """Update index when Education model is updated."""
     index_users.delay([instance.profile.user])
 
 
 @receiver(post_save, sender=Employment, dispatch_uid="employment_post_save_index")
-def handle_update_employment(sender, instance, **kwargs):  # pylint: disable=unused-argument
+def handle_update_employment(sender, instance, **kwargs):
     """Update index when Employment model is updated."""
     index_users.delay([instance.profile.user])
 
 
 @receiver(post_delete, sender=Profile, dispatch_uid="profile_post_delete_index")
-def handle_delete_profile(sender, instance, **kwargs):  # pylint: disable=unused-argument
+def handle_delete_profile(sender, instance, **kwargs):
     """Update index when Profile model instance is deleted."""
     remove_user.delay(instance.user)
 
 
 @receiver(post_delete, sender=Education, dispatch_uid="education_post_delete_index")
-def handle_delete_education(sender, instance, **kwargs):  # pylint: disable=unused-argument
+def handle_delete_education(sender, instance, **kwargs):
     """Update index when Education model instance is deleted."""
     index_users.delay([instance.profile.user])
 
 
 @receiver(post_delete, sender=Employment, dispatch_uid="employment_post_delete_index")
-def handle_delete_employment(sender, instance, **kwargs):  # pylint: disable=unused-argument
+def handle_delete_employment(sender, instance, **kwargs):
     """Update index when Employment model instance is deleted."""
     index_users.delay([instance.profile.user])
+
+
+@receiver(post_save, sender=PercolateQuery, dispatch_uid="percolate_query_save")
+def handle_update_percolate(sender, instance, **kwargs):
+    """When a new query is created or a query is updated, update Elasticsearch too"""
+    index_percolate_queries.delay([instance])
```

Comment:
Ah, I didn't think about the memory issues. I'll pass the id and refactor the other functions to do the same